### ISSUE TITLE
Missing parent in all parse AST nodes

### ISSUE THREAD

**Issue by fpoli** [fpoli/viper-silver#142]
Whenever debugging or creating tests over the parse AST, it is useful and necessary (respectively) to have the 'parent' field properly set. Particularly when assessing the context where the node is within the program.

Some strategies like 'ancestor' provide this information while traversing the AST, in the context object that is carried over during the visit of each node. However, this is an information that should be in the AST, where a field for it already exists (and is set to null at all times, for every node).

**Comment by fpoli** [comment]
> **@a_helfenstein** commented on 2016-05-24 09:52

It even pops the minus out of parentheses:

```
assert (-1 + 1) == 0 // Assertion -(1 + 1) == 0 might not hold.
```

**Comment by fpoli** [comment]
> **@mschwerhoff** commented on 2016-05-24 09:55

**Comment by fpoli** [comment]
> **@seraiah** commented on 2016-05-27 08:48

This is a duplicate of #124

**Comment by fpoli** [comment]
> **@meilers** commented on 2016-07-29 12:50

Resolved in pull request #54.
